__tests__/pages/webhooks/checkout-calculate-taxes.test.ts

```diff
@@ -261,4 +259,181 @@ describe("api/webhooks/checkout-calculate-taxes", () => {
+      .mockReturnValue(
+        Promise.resolve(
+          {} as unknown as joseModule.FlattenedVerifyResult &
+            joseModule.ResolvedKey
+        )
+      );
+    const checkoutPayload = { ...dummyCheckoutPayload };
+
+    const linePayload = checkoutPayload.lines[0];
+    const secondLinePayload = {
+      ...linePayload,
+      id: "Q2hlY2tvdXRMaW5lOjc=",
+      charge_taxes: false,
+    };
+
+    checkoutPayload.lines = [linePayload, secondLinePayload];
+
+    // set mock on next built-in library that build the payload from stream.
+    const rawBodyModule = require("next/dist/compiled/raw-body/index.js");
+    rawBodyModule.default.mockReturnValue({
+      toString: () => JSON.stringify([checkoutPayload]),
+    });
+
+    // set body to undefined as the webhook handler expects that
+    // the processed body doesn't exist.
+    req.body = undefined;
+
+    await calculateTaxes.default(
+      req as unknown as NextApiRequest,
+      res as unknown as NextApiResponse
+    );
+
+    const data: ResponseTaxPayload = res._getData();
+
+    expect(data.lines[0].total_gross_amount).toBe("34.44");
+    expect(data.lines[0].total_net_amount).toBe("28.00");
+    expect(data.lines[0].tax_rate).toBe("0.23");
+
+    expect(data.lines[1].total_gross_amount).toBe("28.00");
+    expect(data.lines[1].total_net_amount).toBe("28.00");
+    expect(data.lines[1].tax_rate).toBe("0");
+
+    expect(res.statusCode).toBe(200);
+
+    mockJose.mockRestore();
+  });
+
+  it("with discounts and line that should not have calculated taxes", async () => {
+    const { req, res } = mockRequest({
+      method: "POST",
+      event: "checkout_calculate_taxes",
+      domain: testDomain,
+      signature:
+        "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6..-Y1p0YWNAuX0kOPIhfjoNoyWAkvRl6iMxWQ",
+    });
+
+    const mockJose = jest
+      .spyOn(joseModule, "flattenedVerify")
+      .mockReturnValue(
+        Promise.resolve(
+          {} as unknown as joseModule.FlattenedVerifyResult &
+            joseModule.ResolvedKey
+        )
+      );
+    const checkoutPayload = { ...dummyCheckoutPayload };
+
+    checkoutPayload.discounts = [{ amount: "2" }, { amount: "1" }];
+    const linePayload = checkoutPayload.lines[0];
+    const secondLinePayload = {
+      ...linePayload,
+      id: "Q2hlY2tvdXRMaW5lOjc=",
+      charge_taxes: false,
+    };
+
+    checkoutPayload.lines = [linePayload, secondLinePayload];
+
+    // set mock on next built-in library that build the payload from stream.
+    const rawBodyModule = require("next/dist/compiled/raw-body/index.js");
+    rawBodyModule.default.mockReturnValue({
+      toString: () => JSON.stringify([checkoutPayload]),
+    });
+
+    // set body to undefined as the webhook handler expects that
+    // the processed body doesn't exist.
+    req.body = undefined;
+
+    await calculateTaxes.default(
+      req as unknown as NextApiRequest,
+      res as unknown as NextApiResponse
```

Comment:
In order to avoid this annoying conversion to `unknown` and then `NextApiRequst` you can do this once in your `mockRequest` function 😉

Check this out:
https://github.com/saleor/react-storefront/blob/main/apps/saleor-app-checkout/test-utils.ts#L11

Reply:
We agree that we will leave this in its current form.